packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2091,10 +2091,16 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     if (widget.obscureText) {
       String text = _value.text;
       text = widget.obscuringCharacter * text.length;
-      final int o =
-        _obscureShowCharTicksPending > 0 ? _obscureLatestCharIndex : null;
-      if (o != null && o >= 0 && o < text.length)
-        text = text.replaceRange(o, o + 1, _value.text.substring(o, o + 1));
+      // Obscure latest character only on mobile
```

Comment:
The code inside the if statement is actually revealing the last character, so maybe this is better:

```
// Reveal the latest character in an obscured field only on mobile.
```

That's assuming that it's true that we want to do this only on mobile.  See my comment about platforms.

Reply:
I have adopted the comment 👍